.github/workflows/ci-asp-app.yaml

```diff
@@ -0,0 +1,133 @@
+      BUILD_PATH:
+        required: true
+        type: string
+      TEMPLATE_PATH:
+        required: true
+        type: string
+      BUILD_FOLDER:
+        required: true
+        type: string
+      PATH_TO_SOLVE_FILE:
+        required: true
+        type: string
+      CSPROJ_FILE:
+        required: true
+        type: string
+      APP_NAME:
+        required: true
+        type: string
+      AZURE_VAULT:
+        required: true
+        type: string
+    secrets:
+      AZURE_CLIENT_ID:
+        required: true
+      AZURE_CLIENT_SECRET:
+        required: true
+      AZURE_TENANT_ID:
+        required: true
+      publish-profile:
+        required: true
+        
+jobs:
+  
+  # Installing dependencies and building the application
+  build:
+    runs-on: windows-latest
+    
+    # Permissions for fetch an OIDC token for a workflow
+    # More information for connect your workflows to authenticate with Azure.
+    # https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-azure
+
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
+      
+      - name: Setup NuGet
+        uses: NuGet/setup-nuget@v1.0.5
+      
+      - name: Restore NuGet packages
+        run: cd ${{ inputs.BUILD_PATH }} && nuget restore ${{ inputs.PATH_TO_SOLVE_FILE }}
+      
+      - name: Download Rattus
+        uses: robinraju/release-downloader@v1.2
+        with:
+          repository: "saritasa/rattus"
+          tag: "0.2"
+          fileName: "rattus-windows-amd64.exe"
+      
+      - name: Render config file
+        run: .\rattus-windows-amd64.exe > ${{ inputs.BUILD_PATH }}/Web.Debug.config
+        env:
+          AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+          AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+          AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+          AZURE_VAULT: ${{ inputs.AZURE_VAULT }}
+          TEMPLATE_PATH: ${{ inputs.TEMPLATE_PATH }}
+     
+      - name: Build app for release
+        run: msbuild ${{ inputs.BUILD_PATH }}/${{ inputs.CSPROJ_FILE }} /p:Configuration=Debug /p:Platform=AnyCPU /t:WebPublish /p:WebPublishMethod=FileSystem /p:DeleteExistingFiles=True /p:publishUrl=${{ env.BUILD_FOLDER }}/${{ env.APP_NAME }}-${{ env.environment }}
+      
+      - name: Upload artifact for deployment job
+        uses: actions/upload-artifact@v2
+        with:
+          name: ASP-app
+          path: ${{ inputs.BUILD_PATH }}/${{ inputs.BUILD_FOLDER }}/${{ inputs.APP_NAME }}-${{ inputs.environment }}
+      
+  # Generated by Azure Depoloyment Center
+  # Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+  # More GitHub Actions for Azure: https://github.com/Azure/actions
+  
+  # Deploying an application to azure cloud
+  deploy:
```

Comment:
Mb it's better to use step with condition instead separated job?

Reply:
It seems to me more convenient when the assembly and deployment are in different jobs, it is convenient to look at the logs in the gh

Reply:
Combined jobs build and deploy